yogstation/code/game/donator/borg_skins.dm

```diff
@@ -0,0 +1,327 @@
+//GLOBAL_VAR_INIT(DonorBorgHolder, /datum/borg_skin_holder) Tried doing it this way, didn't work. :(
+SUBSYSTEM_DEF(YogFeatures)
+	name = "Yog Features" //Kmc's feature dump subsystem
+	flags = SS_BACKGROUND
+	priority = 10
+	var/datum/borg_skin_holder/DonorBorgHolder //Holder datum for borg skins
+
+/datum/controller/subsystem/YogFeatures/fire(resumed = 0) //Runtime avoidance / Anti-Deleting donors
+	if(!DonorBorgHolder)
+		DonorBorgHolder = new /datum/borg_skin_holder
+		return
+
+/datum/borg_skin_holder
+	var/name = "Donator Borg Skin Datumbase"
+	var/datum/borg_skin/skins = list()
+
+/datum/borg_skin_holder/New()
+	generate_borg_skins()
+	. = ..()
+
+/datum/borg_skin_holder/proc/generate_borg_skins()
+	for(var/L in subtypesof(/datum/borg_skin))
+		var/datum/borg_skin/Bskin = L
+		var/datum/borg_skin/instance = new Bskin
+		skins += instance
+	for(var/S in subtypesof(/datum/ai_skin))
+		var/datum/ai_skin/Bskin = S
+		var/datum/ai_skin/instance = new Bskin
+		skins += instance
+
+/datum/borg_skin_holder/proc/AddSkin(var/datum/borg_skin/B)
+	if(!B in skins)
+		skins += B
+		log_game("Successfully added the [B.name] donor borg skin to the datumbase!")
+
+/datum/borg_skin/New()
+	if(SSYogFeatures.DonorBorgHolder)
+		if(!src in SSYogFeatures.DonorBorgHolder.skins)
+			SSYogFeatures.DonorBorgHolder.AddSkin(src) //On new, add the skin to the borg skin database
+
+/mob/living/silicon/robot/proc/PickBorgSkin(var/forced = 0) //We'll do our own AI version inside its pre existent skin selector methinks
+	icon = initial(icon) //Redundancy in case they repick a skin after modulechange
+	if(!SSYogFeatures.DonorBorgHolder)
+		message_admins("[client.ckey] just tried to change their borg skin, but there is no borg skin holder datum! (Has the game not started yet?)")
+		to_chat(src, "An error occured, if the game has not started yet, please try again after it has. The admins have been notified about this")
+		return FALSE
+	if(forced || is_donator(client))//First off, are we even meant to have this verb? or is an admin bruteforcing it onto a non donator for some reason?
+		if(module.name == "Default")
+			to_chat(src, "Please choose a module first! (Standard works too)")
+			return FALSE
+		var/datum/borg_skin/skins = list()
+		for(var/datum/borg_skin/S in SSYogFeatures.DonorBorgHolder.skins)
+			if(S.owner == client.ckey || !S.owner) //We own this skin.
+				if(!S.module_locked || S.module_locked == module.name)
+					skins += S //So add it to the temp list which we'll iterate through
+		var/datum/borg_skin/A //Defining A as a borg_skin datum so we can pick out the vars we want and reskin the unit
+		A = input(src,"Here's a list of your available silicon skins, pick one! (To reset your choice, get a module reset)", "Donator silicon skin picker 9000", A) as null|anything in skins//Pick any datum from the list we just established up here ^^
+		if(!A)
+			return FALSE
+		if(A.name == "Cancel")
```

Comment:
Do you need this? Doesn't 'null|anything in skins' provide a cancel button which will just return null?

Reply:
If the list ever had 1 or fewer items in it, the input failed to display, this was a way to ensure there was always more than one (assuming they had a borg skin available)